### PR TITLE
feat: set default probe for each druid components

### LINKED ISSUE #55: [proposal] Setup default probe for each nodes types
## Context & goal

All the druid components deployed by the operator are deployed by default without probe.
IMHO an operator have to configure in advance this kind of settings.  

Probes have to be configured for each node type and override if an user has defined it.  

all the different probes have to use the [druid api reference](https://druid.apache.org/docs/latest/operations/api-reference.html)

### coordinator, overlord, middlemanager and router

```yaml
      livenessProbe:
        httpGet:
          path: /status/health
          port: $druid.port
        failureThreshold: 20
        initialDelaySeconds: 5
        periodSeconds: 10
        successThreshold: 1
        timeoutSeconds: 5
      readinessProbe:
        httpGet:
          path: /status/health
          port: $druid.port
        failureThreshold: 10
        initialDelaySeconds: 5
        periodSeconds: 10
        successThreshold: 1
        timeoutSeconds: 5
```

### broker

```yaml
      livenessProbe:
        httpGet:
          path: /status/health
          port: $druid.port
        failureThreshold: 20
        initialDelaySeconds: 5
        periodSeconds: 10
        successThreshold: 1
        timeoutSeconds: 5
      readinessProbe:
        httpGet:
          path: /druid/broker/v1/readiness
          port: $druid.port
        failureThreshold: 10
        initialDelaySeconds: 5
        periodSeconds: 10
        successThreshold: 1
        timeoutSeconds: 5
```

### historical

```yaml
      livenessProbe:
        httpGet:
          path: /status/health
          port: $druid.port
        failureThreshold: 20
        initialDelaySeconds: 5
        periodSeconds: 10
        successThreshold: 1
        timeoutSeconds: 5
      readinessProbe:
        httpGet:
          path: /status/health
          port: $druid.port
        failureThreshold: 10
        initialDelaySeconds: 5
        periodSeconds: 10
        successThreshold: 1
        timeoutSeconds: 5
      startUpProbes:
        httpGet:
          path: /druid/historical/v1/loadstatus
          port: $druid.port
        failureThreshold: 20
        initialDelaySeconds: 180
        periodSeconds: 30
        successThreshold: 1
        timeoutSeconds: 10
      
```

### PR DESCRIPTION
**Pull request by cyril-corbon**
Fixes #55 

### Description

Add function to setup default probe properly for each node types 

<hr>

This PR has:
- [x] been tested on a real K8S cluster to ensure creation of a brand new Druid cluster works.
- [ ] been tested for backward compatibility on a real K*S cluster by applying the changes introduced here on an existing Druid cluster. If there are any backward incompatible changes then they have been noted in the PR description.
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added documentation for new or modified features or behaviors.

<hr>

##### Key changed/added files in this PR
 * `MyFoo`
 * `OurBar`
 * `TheirBaz`

### CODE REVIEW

**Comment by cyril-corbon** [comment]
I'm planning to finish this during the week

File: controllers/druid/handler.go [hunk 1]
```diff
@@ -1365,6 +1377,40 @@ func makePodSpec(nodeSpec *v1alpha1.DruidNodeSpec, m *v1alpha1.Druid, nodeSpecUn
...
+			},
+		},
+		InitialDelaySeconds: 5,
+		TimeoutSeconds:      5,
+		PeriodSeconds:       10,
+		SuccessThreshold:    1,
+		FailureThreshold:    10,
+	}
+
+	if nodeType == "historical" && probeType != "liveness" {
```
Comment (itamar-marom) [inline controllers/druid/handler.go:1397]: Just a tiny thing. There are consts for each nodeType you should use them

**Comment by cyril-corbon** [comment]
@itamar-marom @AdheipSingh you can review the doc and the PR. 😄 
I change a bit the values of the probes between the Issue and the PR.
I take those values from our productions.

File: docs/features.md [hunk 2]
```diff
@@ -67,3 +69,103 @@
 - This can be used for init containers or sidecars or proxies etc. 
 - To enable this features users just need to add a new container to the container list 
 - This is scoped at cluster scope only, which means that additional container will be common to all the nodes
+
+## Setup default probe by default
+
+- if you don't define any startup, liveness, readiness probe default probes will be applied.
+- default probe for each components:
+  - coordinator, overlord, middlemanager and router
```
Comment (itamar-marom) [inline docs/features.md:77]: what about historical, query?

**Comment by itamar-marom** [comment]
Except for the little above comment, it all looks great!

**Review by itamar-marom — DISMISSED** [review]

**Comment by itamar-marom** [comment]
Hi @cyril-corbon 
Until @AdheipSingh will approve - it the testdata changes are enough for testing this PR or do you think we should add more tests for the new function?

**Review by itamar-marom — COMMENTED** [review]
We need some unit tests for this one

**Review by AdheipSingh — APPROVED** [review]

**Pull request merged**